test/com/google/enterprise/adaptor/database/JdbcFixture.java

```diff
@@ -47,15 +72,60 @@
    * The default escape character is disabled, to match the DQL
    * behavior (and standard SQL).
    */
+  static {
+    Database dbname = Database.H2;
+    String dbdriver = "org.h2.Driver";
+    String dburl = "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DEFAULT_ESCAPE=";
+    String dbuser = "sa";
+    String dbpassword = "";
+
+    String propertiesFile = System.getProperty("build.properties");
+    if (!Strings.isNullOrEmpty(propertiesFile)) {
+      try (FileInputStream fis =
+          new FileInputStream(new File(propertiesFile))) {
+        Properties properties = new Properties();
+        properties.load(new InputStreamReader(fis, Charset.forName("UTF-8")));
+        if (!Strings.isNullOrEmpty(properties.getProperty("test.database"))) {
+          String name = properties.getProperty("test.database");
+          if (name.equalsIgnoreCase(Database.SQLSERVER.toString())) {
+            dbname = Database.SQLSERVER;
+          } else if (name.equalsIgnoreCase(Database.SQLSERVER.toString())) {
+            dbname = Database.ORACLE;
+          } else {
+            dbname = Database.H2;
+          }
+          if (dbname != Database.H2) {
+            dbdriver = properties.getProperty("test.driver");
```

Comment:
This goes back to the DOORWAYS days, but how about "test.jdbc.driver", et al, in case there are other non-JDBC test properties we may want to set in the future.